File: frontend/src/app/dto/event-search-request.ts
export class EventSearchRequest {
  artistName: string;
  country: string;
  city: string;
  street: string;
  zipCode: string;
  venueName: string;
  eventHall: string;
  startTime: string;
  endTime: string;
  category: string;
  nameOfEvent: string;
  pageIndex: number;
  pageSize: number;

  static getEmptyRequest(): EventSearchRequest {
    return {
      artistName: '',
      country: '',
      city: '',
      street: '',
      zipCode: '',
      venueName: '',
      eventHall: '',
      startTime: '',
      endTime: '',
      category: '',
      nameOfEvent: '',
      pageIndex: 0,
      pageSize: 10
    } as EventSearchRequest;
  }
}

File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/EventSearchRequest.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto;

import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

import static org.springframework.format.annotation.DateTimeFormat.ISO.DATE_TIME;

public record EventSearchRequest(String artistName, String country, String city, String street, Integer zipCode, String venueName, String eventHall,
                                 @DateTimeFormat(iso = DATE_TIME) LocalDateTime startTime, @DateTimeFormat(iso = DATE_TIME) LocalDateTime endTime,
                                 String category,
                                 String nameOfEvent,
                                 int pageIndex,
                                 int pageSize) {
}